redwood-protocol-guest/src/commonMain/kotlin/app/cash/redwood/protocol/guest/ProtocolWidgetSystemFactory.kt

```diff
@@ -15,12 +15,23 @@
  */
 package app.cash.redwood.protocol.guest
 
+import app.cash.redwood.Modifier
+import app.cash.redwood.RedwoodCodegenApi
+import app.cash.redwood.protocol.ModifierTag
 import app.cash.redwood.widget.WidgetSystem
+import kotlinx.serialization.KSerializer
 
 public interface ProtocolWidgetSystemFactory {
   /** Create a new [WidgetSystem] connected to a host via [guestAdapter]. */
   public fun create(
     guestAdapter: GuestProtocolAdapter,
     mismatchHandler: ProtocolMismatchHandler = ProtocolMismatchHandler.Throwing,
   ): WidgetSystem<Unit>
+
+  @RedwoodCodegenApi
+  public fun modifierTag(element: Modifier.Element): ModifierTag
+
+  /** Returns null if the modifier is stateless and should serialize as null. */
+  @RedwoodCodegenApi
+  public fun <T : Modifier.Element> modifierSerializer(element: T): KSerializer<T>?
```

Comment:
On my first attempt this returned a non-null `KSerializer<T>`, but I ran into a [this bug](https://github.com/Kotlin/kotlinx.serialization/pull/2712) in Kotlinx.serialization. I’d like to contribute a fix for that bug and then later I can change this to return non-null always.